src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageSolutionDetailControlModel.cs

```diff
@@ -612,7 +612,7 @@ private void BatchUpdateIsSelectedBasedOnInstalledVersion()
             {
                 foreach (PackageInstallationInfo project in Projects)
                 {
-                    project.IsSelected = project.InstalledVersion != null;
+                    project.IsSelected = project.InstalledVersion != null && project.PackageLevel.Equals(PackageLevel.TopLevel);
```

Comment:
Protect against null PackageLevel.
```suggestion
                    project.IsSelected = project.InstalledVersion != null && PackageLevel.TopLevel.Equals(project.PackageLevel);
```

```suggestion
                    project.IsSelected = project.InstalledVersion != null && project.PackageLevel?.Equals(PackageLevel.TopLevel) == true;
```